packages/web3/src/entitlement.ts

```diff
@@ -718,6 +729,44 @@ async function evaluateCustomEntitledOperation(
     })
 }
 
+async function evaluateEthBalanceOperation(
+    operation: CheckOperation,
+    controller: AbortController,
+    provider: ethers.providers.StaticJsonRpcProvider,
+    linkedWallets: string[],
+): Promise<EntitledWalletOrZeroAddress> {
+    const walletBalances = await Promise.all(
+        linkedWallets.map(async (wallet) => {
+            try {
+                const result = await provider.getBalance(wallet)
+                return {
+                    wallet,
+                    balance: result,
+                }
+            } catch (error) {
+                return {
+                    wallet,
+                    balance: ethers.BigNumber.from(0),
+                }
+            }
+        }),
+    )
+
+    const walletsWithAsset = walletBalances.filter((balance) => balance.balance.gt(0))
```

Comment:
nbd but why filter this out here? seems like an unnecessary iteration. Is there a problem calling  acc.add(0)?

Reply:
oh i see. you're creating a condition that you depend on later.